client/src/rpc_cache.rs

```diff
@@ -0,0 +1,169 @@
+use crate::rpc_config::RpcLargestAccountsConfig;
+use crate::rpc_response::RpcAccountBalance;
+use std::collections::HashMap;
+use std::time::{Duration, SystemTime};
+
+#[derive(Debug, Clone)]
+pub struct LargestAccountsCache {
+    size: usize,
+    duration: u64,
+    cache: HashMap<RpcLargestAccountsConfig, LargestAccountsCacheValue>,
+}
+
+#[derive(Debug, Clone)]
+struct LargestAccountsCacheValue {
+    accounts: Vec<RpcAccountBalance>,
+    cached_time: SystemTime,
+}
+
+impl LargestAccountsCache {
+    pub fn new(size: usize, duration: u64) -> Self {
+        Self {
+            size,
+            duration,
+            cache: HashMap::new(),
+        }
+    }
+
+    pub fn get_largest_accounts(
+        &self,
+        config: &RpcLargestAccountsConfig,
+    ) -> Option<Vec<RpcAccountBalance>> {
+        self.cache.get(&config).map(|value| {
+            value.cached_time.elapsed().ok().map(|elapsed| {
+                if elapsed < Duration::from_secs(self.duration) {
+                    Some(value.accounts.clone())
+                } else {
+                    None
+                }
+            })?
+        })?
```

Comment:
```suggestion
        self.cache.get(&config).and_then(|value| {
            if let Ok(elapsed) = value.cached_time.elapsed() {
                if elapsed < Duration::from_secs(self.duration) {
                    return Some(value.accounts.clone());
                }
            }
            None
        })
```
I think this is slightly more Rusty and slightly easier to parse than the map-try syntax, but not a big deal.